nodes/transforms/deform.py

```diff
@@ -0,0 +1,216 @@
+                default=100.0,
+                min=0.0, max=100.0,
+                update=updateNode)
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, "mode")
+        if self.mode != 'Taper':
+            layout.prop(self, "angles_mode", expand=True)
+        col = layout.column(align=True)
+        row = col.row(align=True)
+        row.prop(self, "lock_x", toggle=True)
+        row.prop(self, "lock_y", toggle=True)
+
+    def sv_init(self, context):
+        self.inputs.new('VerticesSocket', "Vertices", "Vertices")
+        self.inputs.new('MatrixSocket', 'Origin')
+        self.inputs.new('StringsSocket', "Angle").prop_name = "angle"
+        self.inputs.new('StringsSocket', "Factor").prop_name = "factor"
+        self.inputs.new('StringsSocket', "LowLimit").prop_name = "low_limit"
+        self.inputs.new('StringsSocket', "HighLimit").prop_name = "hi_limit"
+
+        self.outputs.new('VerticesSocket', 'Vertices')
+
+        self.update_mode(context)
+
+    def twist(self, mins, maxs, low_limit, hi_limit, angle, vertex):
+        src_min, src_max = limit(mins[2], maxs[2], low_limit, hi_limit)
+        angle = map_linear(src_min, src_max, -angle/2.0, angle/2.0, vertex[2])
+        if self.angles_mode == 'degrees':
+            angle = radians(angle)
+        matrix = Matrix.Rotation(angle, 4, 'Z')
+        return matrix * vertex
+
+    def bend(self, mins, maxs, low_limit, hi_limit, angle, vertex):
+        if self.angles_mode == 'degrees':
+            angle = radians(angle)
+        x,y,z = tuple(vertex)
+        L = maxs[0] - mins[0]
+        angle = 100.0 * angle / (hi_limit - low_limit)
+        R = L / angle
+
+        src_min, src_max = limit(mins[0], maxs[0], low_limit, hi_limit)
+        dx, dy = 0, 0
+        if x < src_min:
+            dx = (x - src_min) * cos(src_min * angle / L)
+            dy = (x - src_min) * sin(src_min * angle / L)
+            x = src_min
+        if x > src_max:
+            dx = (x - src_max) * cos(src_max * angle / L)
+            dy = (x - src_max) * sin(src_max * angle / L)
+            x = src_max
+        scale = x/L
+
+        phi = scale * angle - pi/2.0
+        rho = R - y
+
+        x1 = rho * cos(phi)
+        y1 = rho * sin(phi) + R
+
+        return Vector((x1 + dx, y1 + dy, z))
+
+    def taper(self, mins, maxs, low_limit, hi_limit, factor, vertex):
+        src_min, src_max = limit(mins[2], maxs[2], low_limit, hi_limit)
+        scale = map_linear(src_min, src_max, 1-factor/2, 1+factor/2, vertex[2])
+        x,y,z = tuple(vertex)
+        return Vector((x*scale, y*scale, z))
+
+    def process(self):
+
+        if not any(output.is_linked for output in self.outputs):
+            return
+
+        vertices_s = self.inputs['Vertices'].sv_get(default=[[]])
+        origins = self.inputs['Origin'].sv_get(default=[Matrix()])
+        angles_s = self.inputs['Angle'].sv_get(default=[[pi/4]])
+        factors_s = self.inputs['Factor'].sv_get(default=[[0.785]])
```

Comment:
defaults aren't necessary to set explicitly, when the socket has a `prop_name` pointing at a `self.<property>`